src/test/polymer/polymer-element-scanner_test.ts

```diff
@@ -0,0 +1,119 @@
+import {assert} from 'chai';
+
+import {Visitor} from '../../javascript/estree-visitor';
+import {JavaScriptParser} from '../../javascript/javascript-parser';
+import {ScannedPolymerElement} from '../../polymer/polymer-element';
+import {PolymerElementScanner} from '../../polymer/polymer-element-scanner';
+
+suite('PolymerElementScanner', () => {
+
+  suite('scan()', () => {
+    let scanner: PolymerElementScanner;
+
+    setup(() => {
+      scanner = new PolymerElementScanner();
+    });
+
+    test('finds polymer elements', async() => {
+      const contents = `Polymer({
+        is: 'x-foo',
+        properties: {
+          a: {
+            type: Boolean,
+            value: 5
+          },
+          b: {
+            type: String,
+            value: function() {
+              return 'test';
+            }
+          },
+          c: {
+            type: Number,
+            readOnly: true
+          },
+          d: {
+            type: Number,
+            computed: '_computeD()'
+          },
+          e: {
+            type: String,
+            notify: true
+          },
+          f: {
+            type: Object,
+            observer: '_observeF'
+          },
+          g: {
+            type: {},
+            computed: '_computeG()',
+            readOnly: false
+          },
+          all: {
+            type: Object,
+            notify: true,
+            readOnly: false,
+            reflectToAttribute: false,
+            observer: '_observeAll'
+          }
+        },
+        observers: [
+          '_anObserver()',
+          '_anotherObserver()'
+        ]
+      });
+      Polymer({ is: 'x-bar' });`;
+
+      const document = new JavaScriptParser({
+                         sourceType: 'script'
+                       }).parse(contents, 'test-document.html');
+      let visit = async(visitor: Visitor) => document.visit([visitor]);
+
+      const features = await scanner.scan(document, visit);
+
+      assert.equal(features.length, 2);
+      assert.instanceOf(features[0], ScannedPolymerElement);
+      assert.instanceOf(features[1], ScannedPolymerElement);
+      assert.deepEqual(features.map(f => f.tagName), ['x-foo', 'x-bar']);
```

Comment:
This deepEqual is better than the assert on `.length` because it asserts the same thing, but will give a much better error message if the assertion ever fails (e.g. instead of `expected 1 to equal 2` you'll get `expected ['x-bar'] to equal ['x-foo', 'x-bar']`. Recommend removing the length assertion.

The instanceOf asserts are fine but IMO elidable, the type system does a fine job of it.